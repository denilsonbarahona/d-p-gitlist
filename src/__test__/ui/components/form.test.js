import React from 'react';
import {mount} from 'enzyme';
import {create} from 'react-test-renderer'
import Form from '../../../shared/components/form';


describe('form',()=>{
    const form = mount(<Form />)

    test('render del form',()=>{
        expect(form.length).toEqual(1)
    })

    test('submit form',()=>{
        const Onsubmit = jest.fn()
        const submitEvent = mount(<Form onSubmit={Onsubmit}/>)
        submitEvent.find('form').simulate('submit')
        expect(Onsubmit).toHaveBeenCalledTimes(1)
    })

    test('create snapshot',()=>{
        const formSnapshot = create(<Form />)
        expect(formSnapshot.toJSON()).toMatchSnapshot()
    })
})