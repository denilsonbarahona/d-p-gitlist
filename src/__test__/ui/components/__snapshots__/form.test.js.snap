// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`form create snapshot 1`] = `<form />`;
